src/OSPSuite.Infrastructure.Import/Core/Mappers/DataSetToDataRepositoryMapper.cs

```diff
@@ -29,6 +29,11 @@ public DataRepository ConvertImportDataSet(IDataSource dataSource, int dataSetIn
          var configuration = dataSource.GetImporterConfiguration();
          var dataRepository = new DataRepository { Name = dataSource.NamesFromConvention().ElementAt(dataSetIndex) };
 
+         foreach (var metaDataDescription in dataSource.EnumerateMetaData().ElementAt(dataSetIndex))
```

Comment:
I don't understand this. DataSetIndex is the index in the EnumeratedMetaData. How does the caller knows about this? I would like to understand why we are passing indexes back and forth instead of object instances

Reply:
Sounds correct... I am doing some refactoring here since indeed feels like there is a missing abstraction.